### ISSUE TITLE
Remove use of Intl.RelativeTimeFormat

### ISSUE THREAD

**Issue by andyg0808** [iFixit/react-commerce#1820]
The polyfill for this is costing us **~165k of minified JS** (even for evergreen browsers), and is only used in one place.

https://github.com/iFixit/react-commerce/blob/106d971e1a2ae0494c14afe251a3ee7fbef294ba/frontend/templates/product/sections/ProductReviewsSection/index.tsx#L319

It's being added here:
https://github.com/iFixit/react-commerce/blob/106d971e1a2ae0494c14afe251a3ee7fbef294ba/frontend/pages/_document.tsx#L16-L19

**Comment by dhmacs** [comment]
My understanding is that the polyfills and their associated JS costs only apply if the browser requires them. As `Intl.RelativeTimeFormat` is now widely supported (https://caniuse.com/?search=Intl.RelativeTimeFormat), I'm curious why you're saying that even evergreen browsers needs it.

**Comment by andyg0808** [comment]
I think I just figured it out. It looks like the `curl` command @danielbeardsley and I copied (from testing mobile loading) had an iPhone-ish user agent, so the polyfill service was sending the full polyfill. The user agent starts `Mozilla/5.0`, and I use Firefox, so it wasn't obvious that it was an iPhone user agent.

Loading the polyfills with the correct user agent returns no code, as expected.

Thanks for prompting me to check that again, @dhmacs!

**Comment by sterlinghirsh** [comment]
Would a user get the big polyfill on an iPhone? Mobile devices are most sensitive to big JS bundle size, so if there's a 160K polyfill on an iPhone, I think we'd want to address that.

**Comment by danielbeardsley** [comment]
I experimented and it seems that most iOS devices (at least iphone 12 Pro and SE, and macbook air) all download the big JS. I think we should drop this singular usage or provide a fallback.

**Comment by dhmacs** [comment]
@danielbeardsley What's the version of Safari we're testing with? According to caniuse, the versions of Safari that do not support `Intl.RelativeTimeFormat` account for 0.65% of global usage. If you're observing most iOS devices download the big bundle, does it mean that polyfill is sending more than what's required?

**Comment by danielbeardsley** [comment]
> does it mean that polyfill is sending more than what's required?

Possibly? I'm sure Polyfill.io isn't perfectly matched with reality, there are hundreds of browser features and hundreds of browser versions.

Here's the spec for this polyfill:
https://github.com/Financial-Times/polyfill-library/blob/master/polyfills/Intl/RelativeTimeFormat/config.toml#L19-L32

Though I can't find the spec for the specific locale includes.

My test was done with chrome's devtools device emulation settings.

Looking into it a bit more, it seems iOS mid-12 and below fails this check and this is shipped the polyfill.

https://iosref.com/ios

**Comment by sterlinghirsh** [comment]
Let's make sure that this is only affecting a small percent of users. If it's actually about 1% then I'm fine to leave it as-is.

**Comment by danielbeardsley** [comment]
I spent a bit of time with GA and evaluating those browser constraints on the polyfill. Turns out the biggest chunk is all of Edge which accounts for 4% of our traffic over the last month. Old versions of safari account for another 0.5%.

So 4.5% of our visitors get an extra 160kb of minified JS. Note that this represents about 6% of our revenue because users with these browsers spend more money on average (it seems).

**Comment by danielbeardsley** [comment]
I wonder what happens if we polyfill the feature (9kb) without asking for a polyfill of the locale information.... If it just renders blank strings? or throws an Exception?

**Comment by danielbeardsley** [comment]
I noticed that caniuse says Edge supports this feature down to v79 so I endeavored to open a pull on polyfill.io only to find this: https://github.com/JakeChampion/polyfill-library/pull/1204

Which says:
> fix edge configs :
> chromium edge is normalised to chrome
> any edge configs with versions higher than 18 have been fixed

Effectively any edge version 18 is now mapped to "chrome" in their config. So, I think this feature isn't polyfilled for common edge versions (only very ancient ones). So we're back down to 0.5% so let's not worry about it.
